e2e/scripts/setup_cluster.sh

```diff
@@ -34,7 +34,19 @@ fi
 readonly KUBERNETES_VERSION="$1"
 readonly NUM_NODES="$2"
 
-SSH_PUBLIC_KEYFILE="${SSH_PUBLIC_KEYFILE:-${SCRIPT_DIR}/id_rsa.pub}"
+SSH_PUBLIC_KEYFILE=${SSH_PUBLIC_KEYFILE:-}
+if [ -z "${SSH_PUBLIC_KEYFILE}" ]; then
+  SSH_KEY_NAME="do-ccm-e2e"
+  SSH_PUBLIC_KEYFILE="id_rsa.pub"
+  SSH_KEY_ID="$(doctl compute ssh-key list -ojson | jq --arg NAME ${SSH_KEY_NAME} '.[] | select(.name == $NAME) | .id')"
+  if [ "${SSH_KEY_ID}" ]; then
+    doctl compute ssh-key delete ${SSH_KEY_ID} -f
+  fi
+  if [ ! -f ${SSH_PUBLIC_KEYFILE} ]; then
+    ssh-keygen -t rsa -N "" -f id_rsa
+  fi
+  doctl compute ssh-key import ${SSH_KEY_NAME} --public-key-file ${SSH_PUBLIC_KEYFILE}
```

Comment:
What's the reason we are importing the key?

FWIW, you should be able to log into master and worker nodes without importing it. At least, that's what I tend to do when debugging.

Reply:
The key needs to be available in the users DO account or else we get these errors:
```
W0722 12:04:11.159615    3872 executor.go:130] error running task "Droplet/master-sfo2.masters.ccme2e.snodo.co" (29s remaining to succeed): POST https://api.digitalocean.com/v2/droplets: 422 redacted are invalid key identifiers for Droplet creation.
W0722 12:04:11.159694    3872 executor.go:130] error running task "Droplet/nodes.ccme2e.snodo.co" (29s remaining to succeed): POST https://api.digitalocean.com/v2/droplets: 422 redacted are invalid key identifiers for Droplet creation.
```

Reply:
Ah of course, I completely forgot that I added my key a long long time ago. 🤦‍♂